packages/app/screens/landing/SignUpScreen.js

```diff
@@ -0,0 +1,332 @@
+import React, { useCallback, useRef } from "react";
+import axios from "axios";
+import {
+  StyleSheet,
+  View,
+  Text,
+  KeyboardAvoidingView,
+  TouchableWithoutFeedback,
+  Keyboard,
+  Alert,
+  TouchableOpacity,
+  Platform,
+} from "react-native";
+
+import * as WebBrowser from "expo-web-browser";
+
+import Button from "@app/components/Button";
+import Screen from "@app/components/Screen";
+import TextInput from "@app/components/TextInput";
+import GoogleIcon from "@app/assets/google-icon";
+
+import { useUserContext } from "@app/utils/UserContext";
+import { ENDPOINT } from "@app/utils/constants";
+import { getUserInfo } from "@app/utils/datalayer";
+
+import EyeOff from "@app/assets/eye-line-off";
+import EyeOn from "@app/assets/eye-line-on";
+import * as SecureStore from "@app/utils/SecureStore";
+
+import Constants from "expo-constants";
+import DividerWithText from "@app/components/DividerWithText";
+
+WebBrowser.maybeCompleteAuthSession();
+
+function SignUpScreen({ navigation, route }) {
+  const { user, setUser } = useUserContext();
+
+  const register = async () => {
+    console.log(
+      `Attempting Register with ${inputs.email} and ${inputs.password} at ${ENDPOINT}/auth/local/register`
+    );
+    try {
+      const response = await axios.post(`${ENDPOINT}/auth/local/register`, {
+        firstName: inputs.firstName,
+        lastName: inputs.lastName,
+        email: inputs.email,
+        avatar: "avatar",
+        password: inputs.password,
+      })
+
+      if (response?.data.status == "success") {
+        navigation.navigate("LandingScreen", { email: inputs.email });
+      }
+    } catch (error) {
+      console.log(error.toString());
+    }
+  };
```

Comment:
The logic in your register function is good here, but you are missing the case where the server finds an existing user with the same email and returns a fail type response.
```JSON
{
  "status": "fail",
  "data": {
    "email": "A user with this email already exists.",
  },
}
```

Reply:
We will most likely be implementing this with an alert dialog to show the user when they input an email that is tied to an existing user. I'll have to talk to Napoleon about possibly adding a third-party React Native modal popup/alert dialog library. For now, you can handle this case with a console.log just to make sure it works.